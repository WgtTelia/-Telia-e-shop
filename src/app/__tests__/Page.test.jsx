import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Home from '@/app/page';
import productData from '@/data/products.json';

describe('Page', () => {
  it('renders a heading', () => {
    render(<Home />);

    const heading = screen.getByRole('heading', { level: 1 });

    expect(heading).toBeInTheDocument();
  });

  it('renders product images correctly', () => {
    render(<Home />);
    productData.forEach((product) => {
      const imageElement = screen.getByAltText(product.model);
      expect(imageElement).toBeInTheDocument();
    });
  });

  it('renders product images with correct alt text', () => {
    render(<Home />);
    productData.forEach((product) => {
      const imageElement = screen.getByAltText(product.model);
      expect(imageElement).toHaveAttribute('alt', product.model);
    });
  });
});
