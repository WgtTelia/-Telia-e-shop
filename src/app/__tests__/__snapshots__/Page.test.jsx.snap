// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders homepage unchanged 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-between p-24"
  >
    <div>
      <h1>
        Home
      </h1>
      <a
        href="/about"
      >
        About
      </a>
      <div
        class="bg-white p-6"
      >
        <div
          class="mb-6 rounded-md bg-grey-100 px-4 py-2 align-middle outline outline-2 outline-grey-200"
        >
          <div
            class="align-center flex flex-row items-center justify-center gap-3"
          >
            <div
              class="outline-3 h-[8px] w-[8px] rounded-lg bg-success outline outline-success-light"
            />
            <p
              class="text-success"
            >
              In stock
            </p>
          </div>
        </div>
        <button
          class="rounded-full bg-primary px-4 py-2 font-bold text-white"
        >
          test button
        </button>
      </div>
    </div>
  </main>
</div>
`;
